templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -20,7 +20,7 @@ contents:
     {{end -}}
 
     # If $DHCP6_FQDN_FQDN is not empty and is not localhost.localdomain
-    [[ -n "$DHCP6_FQDN_FQDN" && "$DHCP6_FQDN_FQDN" != "localhost.localdomain" && "$DHCP6_FQDN_FQDN" =~ "." ]] && hostnamectl set-hostname --static --transient $DHCP6_FQDN_FQDN
+    [[ "$DHCP6_FQDN_FQDN" =~ "." && ! "$DHCP6_FQDN_FQDN" =~ localhost ]] && hostnamectl set-hostname --static --transient "${DHCP6_FQDN_FQDN/.*}"
```

Comment:
This worked where I tested it locally, although I'm not sure its correct, it forces the setting on the hostname to the short version, but I'm starting to think we used to expect it to be the long version.